modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamWriter.java

```diff
@@ -268,8 +268,16 @@ public void writeStartElement(String localName) throws XMLStreamException {
                     }
                     if (!queObj.getName().equals(localName) && queObj.getMinOccurs() == 0 &&
                         (stack.isEmpty() || !stack.peek().getName().equals(localName))) {
-                        skipChildElementsFromQueue(queObj);
-                        queObj = queue.peek();
+                        int queueSize = queue.size();
+                        int count = 0;
+                        while (!queObj.getName().equals(localName)) {
+                            if (queue.isEmpty() || count > queueSize) {
```

Comment:
yes, that's the expected behavior. I have added that logic for stopping criteria.